src/components/AddTokenItem.vue

```diff
@@ -2,57 +2,108 @@
     <hr class="hr" />
     <div class="card-flex-container">
         <div style="flex: 1; width: 70px;">
-            <img :src="getImageUrl('logo.png')"
-                style="width: 70px;" />
+            <img :src="getImageUrl('logo.png')" style="width: 70px;" />
         </div>
         <div style="flex: 3 1 0%">
             <p class="bold">{{ props.token.entity.name }}</p>
             <p class="bold">{{ props.token.entity.artist }}</p>
         </div>
-        <el-button :icon="in_playlist ? 'close' : 'plus'" :type="in_playlist ? 'danger' : 'success'" plain circle @click="toggleTokenMeta" :loading="toggle_token_loading"/>
+        <el-button :icon="getIcon" :type="getType" plain circle @click="toggleTokenMeta"
+            :loading="toggle_token_loading" />
     </div>
 
 </template>
 
 <script setup lang="ts">
 
-import {computed, ref } from "vue";
+import { computed, ref, watch } from "vue";
 import type { FirestoreDocument, TokenMeta, Plaque } from "../types/types";
 import { usePlaqueStore } from "@/stores/plaque"
 import { updatePlaque } from "@/api/plaque"
 import { defaultMaxListeners } from "events";
+import { pipelineTopicExpression } from "@babel/types";
 
 interface AddTokenItemProps {
     token: FirestoreDocument<TokenMeta>;
     plaque_id: string;
+    submit_new_token: boolean;
+    clear: string[];
 }
 const props = defineProps<AddTokenItemProps>();
 const toggle_token_loading = ref(false);
-
+const new_tmp_list: string[] = [];
+const new_list = ref(new_tmp_list);
```

Comment:
Are you doing it this way so you can keep the type? Its better to do
`const new_list = ref<string[]>(new_tmp_list);`